src/main/java/com/checkout/ApiClientImpl.java

```diff
@@ -30,7 +30,10 @@
     private final Transport transport;
 
     public ApiClientImpl(final CheckoutConfiguration configuration) {
-        this(new GsonSerializer(), new ApacheHttpClientTransport(configuration.getUri(), configuration.getApacheHttpClientBuilder()));
+        this(new GsonSerializer(),
+                new ApacheHttpClientTransport(configuration.getUri(),
+                        configuration.getApacheHttpClientBuilder(),
+                        configuration.getExecutor()));
```

Comment:
What do you think about refactoring the `ApacheHttpClientTransport` constructor to receive only the `CheckoutConfiguration`?